src/Chess.php

```diff
@@ -334,7 +334,7 @@ protected function makeMove(Move $move): void
         }
 
         // reset the 50 move counter if a pawn is moved or piece is captured
-        if ($move->piece === Piece::PAWN) {
+        if ($move->piece->type === Piece::PAWN) {
```

Comment:
what about `->isPawn()`?

Reply:
that's the idea! I only wanted to go ahead with smaller steps, to minimize diffs (and so to easy reviews in PRs)